app/src/main/java/com/keylesspalace/tusky/components/timeline/viewmodel/CachedTimelineRemoteMediator.kt

```diff
@@ -103,8 +102,12 @@ class CachedTimelineRemoteMediator(
                 val overlappedStatuses = replaceStatusRange(statuses, state)
 
                 if (loadType == LoadType.REFRESH && overlappedStatuses == 0 && statuses.isNotEmpty() && !dbEmpty) {
+                    /* This overrides the last of the newly loaded statuses with a placeholder
+                       to guarantee the placeholder has an id that exists on the server as not all
+                       servers handle client generated ids as expected
+                     */
```

Comment:
What happens when there's only one status?

Reply:
When the timline is loaded, and a refresh returns only one status, and the status is not known already, it would turn into a placeholder at the top of the timeline. Clicking it would load the gap (that one status) normally. (If the one status is known, or the timeline is empty, no placeholder shows)
The Api returning only one unknown status is almost impossible though, because for that you need to already have a placeholder at the top of the timeline, which you can only get by having a placeholder at the usual 30th position, having all 29 statuses before it deleted on the server, and then refreshing (Having everything filtered doesn't count because the filtered statuses are technically still in the local database).  

Not sure if this is 100% the right behavior, maybe it should show the actual status in this case and not a placeholder. I think I might need to test this, thanks for pointing it out.

Reply:
I think the check wheter to insert a placeholder or not must be "we got a full page with no overlap" instead of "we got something with no overlap". Works perfect in my tests, but maybe I am missing another edgecase now 🤔

Reply:
That makes sense to me.
Is there any reason the server returns less than a full page other than "there's nothing else to load in this range" ?

Reply:
I think that depends on the software and the particular timeline you're fetching. In mainline Mastodon, this should currently be fine, but I wouldn't expect that to never change, as there are codepaths (not for timelines as far as I can tell) where we fetch from database first, and do some filtering on top of that, so we may end up with less posts than the page's limit, while there are still underlying posts that weren't returned.

Reply:
D:

Reply:
We have been detecting the end of timeline like that for ages tho

Reply:
For detecting end of timelines, a more reliable way could probably be to check `next` or `prev` headers. But in any case, the current Mastodon behavior is safe, so changing that is not urgent.

Reply:
For the record, I found this commit because I recently experienced “missing chunks” of my timeline, which [I identified](https://social.sitedethib.com/@Claire/108083719157658964) to be Mastodon returning empty pages to Tusky queries mentioning invalid IDs. So that most definitely sounded like #2382 and I expected the fix to that issue to also fix the “missing chunks” issue.

However, I have seen [a report](https://eldritch.cafe/@AgathaSorceress/108089564990952239) of a similar issue from someone using a self-built Tusky that should include that fix. I am myself still running Tusky 16.0, so I have not been able to investigate the issue, but it is possible that there are still issues with the fetching logic.